### PR TITLE
tests: update python dependencies

### PR DESCRIPTION
**Pull request by jcsp**
## Problem

Existing dependencies didn't work on Fedora 39 (python 3.12)

## Summary of changes

- Update pyyaml 6.0 -> 6.0.1
- Update yarl 1.8.2->1.9.4
- Update the `dnf install` line in README to include dependencies of python packages (unrelated to upgrades, just noticed absences while doing fresh pysync run)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: poetry.lock
Comment (bayandin) [inline poetry.lock:1]: > Update pyyaml 6.0 -> 6.0.1
> Update yarl 1.8.2->1.9.4

Those are transitive dependencies, and we don't need to update anything in `pyproject.toml`?

File: README.md
Comment (bayandin) [inline README.md:1]: While we're here, could you please update `python-poetry` to `python3-poetry` for Debian-based instruction (`python-poetry` doesn't exist)?

I've noticed it some time ago, but didn't want to open a PR just for these changes :|

Comment (jcsp) [reply README.md:1]: Hmm, I'm on Fedora so can't really test any changes to debian instructions -- the dependencies may also be out of date for debian.  Probably makes sense to open a separate PR to freshen the debian instructions (by someone who can check it's correct).

File: poetry.lock
Comment (jcsp) [reply poetry.lock:1]: Right.  I had to edit the lock file by hand to make this work: `poetry` itself is hanging if I try and use `poetry update` or similar (or just extremely slow, waited 10+ minutes)

File: README.md
Comment (jcsp) [reply README.md:1]: Okay - I you want to open a suggestion I don't mind clicking commit on it

Comment (bayandin) [reply README.md:1]: Pushed a commit.
GitHub didn't allow to add a suggestion for it (too far from existing changes in the PR 🤷 )

**Review by bayandin — APPROVED** [review]
Thanks John!

**Review by tristan957 — APPROVED** [review]

**Pull request merged**